test/threadstest.c

```diff
@@ -1358,6 +1358,42 @@ static int test_x509_store(void)
     return ret;
 }
 
+/* Test using OBJ_create in multiple threads */
+static void test_obj_create_worker(void)
+{
+    int i, nid, nid2;
+    time_t now;
+    char name[40];
+
+    for (i = 0; i < 4; i++) {
+        now = time(NULL);
+        sprintf(name, "Time in Seconds = %ld", (long) now);
+        while (now == time(NULL))
+            /* no-op */;
```

Comment:
What is this loop doing here?  I know it's waiting for the next second but why?

So all threads kick off together?  That should be instigated in `thread_run_test()` if possible.

None of the other threading tests seem to need this.

Reply:
I can only test very few name collisions here, because otherwise the hash table would create
a huge memory leak.  So I had to slow down the test very much.  I think its a special case...